harness/adapters/google.py

```
@@ -160,9 +169,40 @@ def _build_thinking_config(self, budget: int) -> Optional[Any]:
+            "high": 8192,
+        }
+        normalized = value.strip().lower()
+        if normalized in effort_map:
+            return effort_map[normalized]
+        try:
+            return int(normalized)
+        except ValueError:
+            raise ValueError(
+                f"Unsupported thinking_budget value: {value!r}. "
+                f"Expected 'low', 'medium', 'high', or a numeric string."
+            )
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)